### PR TITLE
incoming webhook trigger doesn't work with gitlab,

### PR DESCRIPTION
**Pull request by rcerven**
Repository CR, needs spec->git_provider->type specified for anything what isn't github app

[STONEBLD-2681](https://issues.redhat.com//browse/STONEBLD-2681)

### Checklist:
 - PR has reference to the issue(s) it resolves
 - Write / update unit tests
 - Write / update integration (envtest) tests
 - Ensure there is an issue for e2e tests if needed
 - Ensure `make test` passes
 - Ensure test coverage hasn't decreased
 - Test code changes manually
 - Update readme and documentation
 - Write PR description that highlights overall changes and add usage examples if applicable

### CODE REVIEW

File: controllers/component_build_controller_pac_repository.go [hunk 1]
```diff
@@ -225,6 +225,14 @@ func generatePACRepository(component appstudiov1alpha1.Component, config *corev1
 		gitProviderConfig.URL = url
 	}
 
+	if gitProviderConfig != nil {
+		repositoryGitProvider := gitProvider
+		if repositoryGitProvider == "bitbucket" {
+			repositoryGitProvider = "bitbucket-cloud"
```
Comment (chmeliik) [inline controllers/component_build_controller_pac_repository.go:231]: Could you add a comment with a link to some documentation that would explain why `bitbucket-cloud` is the correct value?

File: controllers/component_build_controller_unit_test.go
Comment (chmeliik) [inline controllers/component_build_controller_unit_test.go:1]: It looks like we don't have a bitbucket test case

File: controllers/component_build_controller_pac_repository.go (hunk 1 above)
Comment (rcerven) [reply controllers/component_build_controller_pac_repository.go:232]: added

Comment (chmeliik) [reply controllers/component_build_controller_pac_repository.go:232]: nitpick: https://pipelinesascode.com/docs/guide/incoming_webhook/#incoming-webhook-url looks like the more relevant section

File: controllers/component_build_controller_unit_test.go
Comment (chmeliik) [reply controllers/component_build_controller_unit_test.go:1]: Discussed offline: bitbucket isn't fully supported, no need to add a testcase yet

**Review by chmeliik — APPROVED** [review]

File: controllers/component_build_controller_pac_repository.go (hunk 1 above)
Comment (rcerven) [reply controllers/component_build_controller_pac_repository.go:232]: updated

**Review by mmorhun — APPROVED** [review]

File: controllers/component_build_controller_pac_repository.go (hunk 1 above)
Comment (savitaashture) [reply controllers/component_build_controller_pac_repository.go:232]: @rcerven naive query 
may be i m missing something 
PAC already support bitbucket-cloud as per the doc https://pipelinesascode.com/docs/guide/incoming_webhook/#incoming-webhook-url any reason for overriding it :thinking:

Comment (rcerven) [reply controllers/component_build_controller_pac_repository.go:232]: @savitaashture PAC requires 'bitbucket-cloud' string, but in build-service we are using 'bitbucket' that is why I am converting it to string which PAC understands

Comment (savitaashture) [reply controllers/component_build_controller_pac_repository.go:232]: Okay got it thank you

**Pull request merged**